python/pyspark/tests.py

```diff
@@ -874,6 +874,15 @@ def test_sortByKey_uses_all_partitions_not_only_first_and_last(self):
             for size in sizes:
                 self.assertGreater(size, 0)
 
+    def test_pipe_functions(self):
+        data = ['1', '2', '3']
+        rdd = self.sc.parallelize(data)
+        with QuietTest(self.sc):
+            self.assertRaises(Py4JJavaError, rdd.pipe('cc').collect)
+        result = rdd.pipe('cat').collect()
```

Comment:
@davies, I guess that your comment about `grep` is due to the fact that the fix here might break programs that used to work in case they relied on the old behavior? I guess that one option might be to add a configuration flag for controlling the exit-code-handling behavior and to update the documentation to address this. Any thoughts?